src/main/java/reposense/git/GitRevList.java

```diff
@@ -57,15 +57,28 @@ public static String getCommitHashInRange(String root, String branchName, String
             return startHash;
         }
 
-        Path rootPath = Paths.get(root);
-        String revListCommand = "git rev-list " + startHash + ".." + endHash + " " + branchName
-                + REVISION_PATH_SEPARATOR;
-        String output = runCommand(rootPath, revListCommand);
+        String fromStartHash = getAllCommitHashSince(root, branchName, startHash);
+        String fromEndHash = getAllCommitHashSince(root, branchName, endHash);
+        StringBuilder output = new StringBuilder();
 
-        if (output.equals("")) {
-            return endHash + "\n" + startHash;
+        if (fromStartHash.length() > fromEndHash.length()) {
+            output.append(fromStartHash.substring(fromEndHash.length()));
```

Comment:
should this be `fromStartHash.substring(0, fromStartHashLength - fromEndHashLength)` instead?

Reply:
The `rev-list` command outputs the list of commits starting from `HEAD` until the given commit, so this part is removing the parts that are common between both `fromStartHash` list and `fromEndHash` list.

For example, if `fromStartHash` has:
```
abc
def
ghi
123
```

And `fromEndHash` has:
```
abc
def
```

Then, I want to get just `ghi` and `123`, since that is the result of the set difference, and that starts from the index which corresponds to the length of `fromEndHash`.

I think it is quite confusing at the first glance, so I have added a comment to explain what it is intending to do.

Reply:
noted, thanks for the clear explanation 👍🏻 abc -> def -> ghi -> 123 sorted based on the most recent commit right

Reply:
Yep, the newest commit is `abc` and the oldest commit is `123`.